cmd/restic/cmd_rewrite_integration_test.go

```diff
@@ -170,5 +171,133 @@ func TestRewriteSnaphotSummary(t *testing.T) {
+}
+
+func TestRewriteInclude(t *testing.T) {
+	env, cleanup := withTestEnvironment(t)
+	defer cleanup()
+	createBasicRewriteRepo(t, env)
+	snapshots := testListSnapshots(t, env.gopts, 1)
+
+	// restic rewrite <snapshots[0]> -i "*.txt" --forget
+	rtest.OK(t, runRewrite(context.TODO(), RewriteOptions{
+		Forget:                true,
+		IncludePatternOptions: filter.IncludePatternOptions{Includes: []string{"*.txt"}}},
+		env.gopts,
+		[]string{snapshots[0].String()}))
+	newSnapshots := testListSnapshots(t, env.gopts, 1)
+	rtest.Assert(t, snapshots[0] != newSnapshots[0], "snapshot id should have changed")
+
+	// read restic ls output and count .txt files
+	out := testRunLsWithOpts(t, env.gopts, LsOptions{}, []string{newSnapshots[0].Str()})
+	fileList := strings.Split(string(out), "\n")
+	count := 0
+	for _, filename := range fileList {
+		if strings.Contains(filename, ".txt") {
+			count++
+		}
+	}
+	rtest.Assert(t, count == 2, "there should be 2 txt files in the snapshot, but there are %d files", count)
+
+	// get snapshot summary and find these 2 files
+	_, repo, unlock, err := openWithExclusiveLock(context.TODO(), env.gopts, false)
+	rtest.OK(t, err)
+
+	sn, err := restic.LoadSnapshot(context.TODO(), repo, newSnapshots[0])
+	rtest.OK(t, err)
+
+	rtest.Assert(t, sn.Summary != nil, "snapshot should have a summary attached")
+	rtest.Assert(t, sn.Summary.TotalFilesProcessed == 2,
+		"there should be 2 files in the snapshot, but there are %d files", sn.Summary.TotalFilesProcessed)
+	unlock()
+}
+
+func TestRewriteEmptyDirectory(t *testing.T) {
+	env, cleanup := withTestEnvironment(t)
+	defer cleanup()
+	createBasicRewriteRepo(t, env)
+	snapshots := testListSnapshots(t, env.gopts, 1)
+
+	// restic rewrite <snapshots[0]> -i empty-directory --forget
+	rtest.OK(t, runRewrite(context.TODO(), RewriteOptions{
+		Forget:                true,
+		IncludePatternOptions: filter.IncludePatternOptions{Includes: []string{"empty-directory"}}},
+		env.gopts,
+		[]string{snapshots[0].Str()}))
+	newSnapshots := testListSnapshots(t, env.gopts, 1)
+	rtest.Assert(t, snapshots[0] != newSnapshots[0], "snapshot id should have changed")
+
+	// read restic ls output and count lines which contain the string "empty-directory"
+	out := testRunLsWithOpts(t, env.gopts, LsOptions{}, []string{newSnapshots[0].String()})
+	fileList := strings.Split(string(out), "\n")
+	count := 0
+	for _, filename := range fileList {
+		if strings.Contains(filename, "empty-directory") {
+			count++
+		}
+	}
+	rtest.Assert(t, count == 1, "there should be 1 empty directory in the snapshot, but there are %d snapshots", count)
+
+	// get snapshot summary and find this 1 directory
+	_, repo, unlock, err := openWithExclusiveLock(context.TODO(), env.gopts, false)
+	rtest.OK(t, err)
+
+	sn, err := restic.LoadSnapshot(context.TODO(), repo, newSnapshots[0])
+	rtest.OK(t, err)
+
+	rtest.Assert(t, sn.Summary != nil, "snapshot should have a summary attached")
```

Comment:
Please additionally use `testRunLsWithOpts` to check that the directory has actually be removed.